### PR TITLE
Add custom claim mapping feature

### LINKED ISSUE #1627: Passing end user attributes to the backend using JWT
### Describe your problem(s)
Backend JWT generation feature should be implemented in microgateway in order to send end user details to the backend.

### Describe your solution
Implement the backend jwt generation feature using the `org.wso2.carbon.apimgt.gateway.common` package.
### How will you implement it

---

### LINKED ISSUE #1673: Add custom claims to back end JWT
### Describe your problem(s)
The jwt generated for the backend should have the ability to change the claims present in the incoming authentication jwt
There should be way to alter the jwt claim key and value both.

### How will you implement it

---
#### Related Issues:
Improvement on the issue: https://github.com/wso2/product-microgateway/issues/1627

### PR DESCRIPTION
**Pull request by ShalkiWenushika**
### Purpose
This PR will add the custom claim mapping feature to the envoy microgateway.

### Issues
Fixes #1627 
Fixes #1673 

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: test/test-integration/src/test/java/org/wso2am/micro/gw/tests/jwtGenerator/JwtTransformerTestCase.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+        JSONObject responseJSON = new JSONObject(response.getData());
+        String tokenFull = responseJSON.get("token").toString();
+        String strTokenBody = tokenFull.split("\\.")[1];
+        String decodedTokenBody = new String(Base64.getUrlDecoder().decode(strTokenBody));
+        JSONObject tokenBody = new JSONObject(decodedTokenBody);
+        Assert.assertEquals(tokenBody.get("CUSTOM-CLAIM"), "admin",
+                "The custom claim has not correctly mapped");
+    }
+}
```
Comment (Rajith90) [inline test/test-integration/src/test/java/org/wso2am/micro/gw/tests/jwtGenerator/JwtTransformerTestCase.java:35]: Could we please add a test case with custom java transformer as well

File: enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/JwtTransformerAnnotation.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ */
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+
+public @interface JwtTransformerAnnotation {
+    boolean enabled() default false;
+    String name();
+    String issuer();
+}
```
Comment (Rajith90) [inline enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/JwtTransformerAnnotation.java:18]: Could you please explain what is the annotation I need to use in my custom transformer. I mean a sample annotation if my issuer name is "issuer1"

File: enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/JWTUtil.java [hunk 3]
```diff
@@ -246,5 +250,50 @@ public static AbstractAPIMgtGatewayJWTGenerator getApiMgtGatewayJWTGenerator() {
...
         return jarFilesList;
     }
+
+    public static Map<String, JWTTransformer> loadJWTTransformers() {
+        JWTTransformer jwtTransformer = null;
+        List<String> jarFilesList = getJarFilesList();
+        Map<String, JWTTransformer> jwtTransformersMap = new HashMap<>();
+        for (int fileIndex = 0; fileIndex < jarFilesList.size(); fileIndex++) {
+            try {
+                String pathToJar = JwtConstants.DROPINS_FOLDER + jarFilesList.get(fileIndex);
```
Comment (Rajith90) [inline enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/JWTUtil.java:260]: Is the dropins folder value hard coded or got from environment variable

Comment (ShalkiWenushika) [reply enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/JWTUtil.java:260]: It is hard coded. It refers to `/home/wso2/lib/dropins/` location.

File: enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/JwtTransformerAnnotation.java (hunk 2 above)
Comment (ShalkiWenushika) [reply enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/JwtTransformerAnnotation.java:21]: Added comments.

File: test/test-integration/src/test/java/org/wso2am/micro/gw/tests/jwtGenerator/JwtTransformerTestCase.java (hunk 1 above)
Comment (ShalkiWenushika) [reply test/test-integration/src/test/java/org/wso2am/micro/gw/tests/jwtGenerator/JwtTransformerTestCase.java:35]: Added a new test case: CustomJwtTransformerTestCase

**Review by Rajith90 — APPROVED** [review]

**Pull request merged**